components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -323,6 +348,140 @@ fn cast_json_as_uint(ctx: &mut EvalContext, val: &Option<Json>) -> Result<Option
     }
 }
 
+// cast any as real, some cast functions reuse `cast_any_as_any`
+//
+// cast_decimal_as_signed_real -> cast_any_as_any<Decimal, Real>
+// cast_time_as_real -> cast_any_as_any<Time, Real>
+// cast_duration_as_real -> cast_any_as_any<Duration, Real>
+// cast_json_as_real -> by cast_any_as_any<Json, Real>
+
+#[rpn_fn]
+#[inline]
+fn cast_signed_int_as_signed_real(val: &Option<Int>) -> Result<Option<Real>> {
+    match val {
+        None => Ok(None),
+        Some(val) => Ok(Real::new(*val as f64).ok()),
```

Comment:
If fail, should we capture the failure with `ctx` ?

Reply:
There is always no err here.(Always not NaN), but we should avoid panic from unwrap.

Reply:
There is always no err here.(Always not NaN), but we should avoid panic from unwrap.